lib/happo/uploader.rb

```diff
@@ -62,9 +62,9 @@ def upload_diffs
     def find_or_build_bucket
       service = S3::Service.new(access_key_id: @s3_access_key_id,
                                 secret_access_key: @s3_secret_access_key)
-      bucket = service.buckets.find(@s3_bucket_name)
+      bucket = service.bucket(@s3_bucket_name)
 
-      if bucket.nil?
+      if !bucket.exists?
         bucket = service.buckets.build(@s3_bucket_name)
```

Comment:
The `proxy_owner` argument in `service.buckets.build` seems to be just `service`, which is passed into BucketExtensions as `self` in `service.buckets` (https://github.com/qoobaa/s3/blob/ec570632d25a3571cd4585de08e48e72b838f048/lib/s3/service.rb#L48). `service.bucket` also creates a bucket with `self` for the same argument (https://github.com/qoobaa/s3/blob/ec570632d25a3571cd4585de08e48e72b838f048/lib/s3/service.rb#L55). Thus, the bucket `service.buckets.build` creates is actually identical to a bucket created by `service.bucket`. I will submit a PR removing this line in `uploader.rb`.